backend/Application/Api/GraphQL/Search/SearchResult.cs

```diff
@@ -98,4 +115,70 @@ public IEnumerable<NodeStatus> GetNodeStatuses([Service] NodeStatusSnapshot node
         return nodeSummarySnapshot.NodeStatuses
             .Where(x => x.NodeName != null && x.NodeName.Contains(_queryString, StringComparison.InvariantCultureIgnoreCase));
     }
+    
+    /// <summary>
+    /// Try match query with contract regex.
+    /// 
+    /// Only consider query if there is a match.
+    /// </summary>
+    internal static bool TryMatchContractPattern(
+        string? query, out string? pattern)
+    {
+        const char end = '%';
+        const char start = '<';
+        const char comma = ',';
+        
+        pattern = null;
+        if (query is null)
+        {
+            return false;
+        }
+        
+        var match = ContractAddressRegex.Match(query);
+        if (!match.Success) return false;
+
+        switch (match.Groups[2].Success)
```

Comment:
I suggest to replace this expression with a variable, which makes it explicit what part of the query was matched in Groups [2]. This will enhance readability, but also require less insight into ContractAddressRegex, to understand what values the different indexes contain in the returned matched list.